File: src/Web/src/model/Device/DeviceModuleDto.ts
import { DeviceModuleModel } from "../../enum/DeviceModuleModel";

export interface DeviceModuleDto {
  guid: string;
  name: string;
  serialNumber: string;
  address: string;
  mac: string;
  port: string;
  firmware: string;
  model: DeviceModuleModel;
  deviceModules: DeviceModuleDto[];
  locationGuid: string;
}

File: src/Modules/Contracts/Core.Contract/DTOs/DeviceModule/DeviceModuleDto.cs
using SharedKernel.Enums;

namespace Core.Contract.DTOs.DeviceModule;

public sealed record DeviceModuleDto(
      Guid Guid,
      string Name,
      string SerialNumber,
      string Firmware,
      string Mac,
      int Port,
      int Address,
      DeviceModuleModel Model,
      int ReaderSlot,
      int OutputSlot,
      int InputSlot,
      Guid DeviceGuid,
      string DeviceName,
      Guid LocationGuid,
      string LocationName,
      bool IsActive,
      bool IsDefault
);